### PR TITLE
Switch range definition in switch configuration

### PR DESCRIPTION
**Pull request by fdurand**
## Description

It's now possible to add a switch range (cird format) for the switch ip address configuration parameter.
It's usefull if on the same network you have multiples switch/ap with the same configuration, you just have to define one.
## Impacts

No impacts
## Delete branch after merge

YES
## NEWS file entries

New Features
+++++++++++
- Define range of network switches in switch configuration

### CODE REVIEW

File: lib/pf/SwitchFactory.pm [hunk 1]
```diff
@@ -79,13 +80,30 @@ sub instantiate {
     });
 
     my $switch_data;
+    my $i = 1;
```
Comment (rouzier) [inline lib/pf/SwitchFactory.pm]: Remove

File: lib/pf/SwitchFactory.pm [hunk 2]
```diff
@@ -79,13 +80,30 @@ sub instantiate {
...
     foreach my $search (@requestedSwitches){
         if($SwitchConfig{$search}){
             $requestedSwitch = $search;
             $switch_data = $SwitchConfig{$search};
+            $i = 0;
```
Comment (jrouzierinverse) [inline lib/pf/SwitchFactory.pm]: Remove

File: lib/pf/SwitchFactory.pm [hunk 3]
```diff
@@ -79,13 +80,30 @@ sub instantiate {
...
             last;
         }
     }
+    if ($i) {
```
Comment (jrouzierinverse) [inline lib/pf/SwitchFactory.pm]: Just check if $requestedSwitch is set

File: lib/pf/util.pm [hunk 4]
```diff
@@ -870,10 +872,20 @@ sub valid_mac_or_ip {
         my ($mac) = clean_mac($mac_or_ip);
         return 1 if($mac && $mac !~ $NON_VALID_MAC_REGEX && $mac =~ $VALID_PF_MAC_REGEX);
     }
-    get_logger()->error("invalid MAC or IP: $mac_or_ip");
+    get_logger()->warn("invalid MAC or IP: $mac_or_ip");
```
Comment (jrouzierinverse) [inline lib/pf/util.pm]: This should be debug to reduce log noise

**Comment by julsemaan** [comment]
An example in conf/switches.conf would be necessary IMO and maybe a note in the admin guide.

_Insert some joke that includes the word zammit here_

**Comment by jrouzierinverse** [comment]
Add a test case in t/SwitchFactory.t

**Comment by julsemaan** [comment]
Code looks good but as discussed it's missing the freeradius part

File: t/SwitchFactory.t [hunk 5]
```diff
@@ -99,6 +99,15 @@ $switch = pf::SwitchFactory->instantiate({ switch_mac => "ff:01:01:01:01:04", sw
 isa_ok( $switch, 'pf::Switch::Cisco::Catalyst_2900XL' );
 is($switch->{_id}, '192.168.0.1', "Proper id is set");
 
+#Test using ip address in a range
+$switch = pf::SwitchFactory->instantiate('192.168.3.1');
+isa_ok($switch, 'pf::Switch::Cisco::Catalyst_2960');
+is($switch->{_id}, '192.168.3.1', "Proper id is set 192.168.3.1");
```
Comment (dwlfrth) [inline t/SwitchFactory.t]: Proper ID is set for

**Comment by jrouzierinverse** [comment]
Please Review and test if dynamic controller is working

**Comment by whitx** [comment]
Tested, the priority apply properly @mac ->@ip -> range IP.

**Pull request merged**